runtime/wasm/src/host.rs

```diff
@@ -570,24 +570,45 @@ impl RuntimeHostTrait for RuntimeHost {
     async fn process_block(
         &self,
         logger: &Logger,
-        block: &Arc<LightEthereumBlock>,
+        block: &Arc<EthereumBlockType>,
         trigger_type: &EthereumBlockTriggerType,
         state: BlockState,
         proof_of_indexing: SharedProofOfIndexing,
     ) -> Result<BlockState, anyhow::Error> {
         let block_handler = self.handler_for_block(trigger_type)?;
+        let mapping_block: EthereumBlockType = match trigger_type {
+            EthereumBlockTriggerType::Every(BlockType::FullWithReceipts) => match self
+                .host_exports
+                .ethereum_adapter
+                .load_full_block(logger, block.light_block().clone())
+                .compat()
+                .await
+            {
+                Ok(block) => Ok(EthereumBlockType::FullWithReceipts(block)),
+                Err(e) => Err(anyhow::anyhow!(
```

Comment:
Consider using `with_context` to add additional information to an error, rather than formatting the error. https://docs.rs/anyhow/1.0.3/anyhow/trait.Context.html#tymethod.with_context